src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,235 @@
+        this.disableButton = this._disableButton.bind(this);
+        this.submit = this._submit.bind(this);
+    }
+
+    getCheckedValue(e) {
+
+        return e.target.checked;
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
+    }
+
+    _fieldBlurred(ev) {
+
+        const isBlurred = { ...this.state.isBlurred };
+        const field = ev.target.id;
+
+        isBlurred[field] = true;
+
+        this.setState({ isBlurred });
+    }
+
+    _showEmailError() {
+
+        return (this.state.isBlurred.email) && !IsEmail(this.state.email);
+    }
+
+    _showPasswordError() {
+
+        return this.state.isBlurred.confirmPassword && (this.state.password !== this.state.confirmPassword);
+    }
+
+    _disableButton() {
+
+        const { firstName, lastName, email, currentPassword, password, confirmPassword } = this.state;
+        // const fieldHasValue = (firstName && lastName && email && password) !== '';
+        const fieldHasChanged =
+            (firstName !== this.props.userDetails.firstName)
+            || (lastName !== this.props.userDetails.lastName)
+            || (email !== this.props.userDetails.email);
+        const passwordsMatch = password === confirmPassword;
+        const requestPasswordChange = (password !== '')
+            && passwordsMatch
+            && (currentPassword !== password)
+            && (currentPassword !== '');
+
+        // if (fieldHasValue && IsEmail(email) && passwordsMatch) {
+        //     return false;
+        // }
+
+        // TODO: Make sure this OR statement doesn't mess things up if something is half-filled or something
+        if (fieldHasChanged || (requestPasswordChange)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    _submit(ev) {
+
+        const { firstName, lastName, email, password } = this.state;
+
+        this.props.onSubmit({ email, password, firstName, lastName  });
+
+        ev.preventDefault();
+    }
+
+    render() {
+
+        return (
+
+            <FormWrapper>
+                <form onSubmit={this.submit}>
+                    <TextWrapper>
+                        <Typography variant='headline' gutterBottom>Settings</Typography>
+                        {this.props.errorMessage &&
+                            <FormHelperText>Error: {this.props.errorMessage}</FormHelperText>
+                        }
+                        <TextField
+                            id='firstName'
+                            type='text'
+                            variant='outlined'
+                            label='First Name'
+                            margin='normal'
+                            fullWidth
+                            value={this.fieldValue('firstName')}
+                            onChange={this.proposeNew('firstName')}
+                            onBlur={this.fieldBlurred}
+                        />
+                        <TextField
+                            id='lastName'
```

Comment:
Does this introduce an actual `id` attribute?  If so, I might suggest going with something else like `name`, or passing the name of the field directly to `fieldBlurred()`.

Reply:
According to the [material docs](https://material-ui.com/api/text-field/#props),

>The id of the input element. Use that property to make label and helperText accessible for screen readers.

I'm not sure if that prop is doing something for me automatically or not, though.

Reply:
Got it!  Yes, carry on! :P